Marlin/src/lcd/marlinui.cpp

```diff
@@ -1400,7 +1403,33 @@ void MarlinUI::init() {
 
     const enc_t live_enc = { BUTTON_PRESSED(EN1), BUTTON_PRESSED(EN2) };
 
-    #if ENCODER_DEBOUNCE_MS
+    #if defined(USE_COUNTER_DEBOUNCING)
+      static uint16_t counter;
+      counter++;
+
+      static int dt_us = 500;  // the time delta in us between each run of this function
+      if (counter == 1000){
+        static millis_t last_ms;
+        dt_us = (now-last_ms);
```

Comment:
Right, this counts the time in ms per 1000 calls. I do this to get a microsecond accuracy average.
So it is 1000 times too many ms, but the right value in microseconds. I do this to avoid both calling micros() and doing divisions

Reply:
Makes sense, thanks.